### PR TITLE
Update missing value handling and placeholder logic

### PR DESCRIPTION
**Pull request by JHogenboom**
- Replaced `na` key with `missing_values` for improved consistency.
- Enhanced logic in `_compute_local_missing_values` to accurately count placeholders while avoiding duplication of structural NaNs.
- Added detailed unit tests to cover edge cases and scenarios involving placeholder replacement.

Fixing #12

### CODE REVIEW

File: src/vantage6_strongaya_general/general_statistics.py [hunk 1]
```diff
@@ -919,35 +919,38 @@ def _compute_local_min_max(
...
     Returns:
         Tuple[int, pd.Series]: The count of missing values and the updated column values.
     """
-    if isinstance(placeholder, int):
-        true_na_count = (column_values == placeholder).sum()
-        if replace_with_na:
-            column_values = column_values.replace(placeholder, pd.NA)
-    elif isinstance(placeholder, str):
-        true_na_count = column_values.eq(placeholder).sum()
+    if placeholder is not None and not pd.isna(placeholder):
```
Comment (JHogenboom) [inline src/vantage6_strongaya_general/general_statistics.py:944]: With handling in v6-tools-rdf not strictly necessary; could cause trouble when called double (counting missing in v6-tools-rdf and then again here, but current flow will skip this function anyway if that is the case: 

```python
        if "missing_values" in column_stats:
            na_count = column_stats["missing_values"]
        else:
            na_count, column_values = safe_calculate(
                _compute_local_missing_values,
```
https://github.com/STRONGAYA/v6-tools-general/pull/17/changes#diff-1115e205f27e0fac03be6ef0a01742ffb40e676b5d0c530714f1b8cd4d8a704fR598-R599

**Review by JHogenboom — COMMENTED** [review]
Looks good to go, the main culprit sat in v6-tools-rdf use of `missing_values` rather than `na`.

Logic with placeholder was updated but remains the same in v6-tools-rdf: 
```python
def add_missing_data_info(df: pd.DataFrame, placeholder: Any) -> None:
    """
    Add missing data information to the DataFrame using the predetermined_info attribute.

    Args:
        df (pd.DataFrame): The DataFrame to add missing data information to.
        placeholder (Any): The placeholder to use for missing data.
    """
    if not hasattr(df, "predetermined_info"):
        df.predetermined_info = PredeterminedInfoAccessor(df)
    df.predetermined_info.add_stat(
        "missing_values",
        calculation_func=_compute_local_missing_values,
        per_column=True,
        store_output_index=0,
        placeholder=placeholder,
    )
```
https://github.com/STRONGAYA/v6-tools-rdf/blob/main/src/vantage6_strongaya_rdf/data_processing.py#L18-L34

File: src/vantage6_strongaya_general/general_statistics.py [hunk 2]
```diff
@@ -595,8 +595,8 @@ def _orchestrate_local_numerical_statistics(
                 pass
 
         # Count the occurrences of missing values safely if it does not exist yet
-        if "na" in column_stats:
-            na_count = column_stats["na"]
+        if "missing_values" in column_stats:
+            na_count = column_stats["missing_values"]
```
Comment (bartvanb) [inline src/vantage6_strongaya_general/general_statistics.py:599]: I don't fully understand this change - I don't see changes that create this `missing_values` column, so when is it ever present?

Comment (bartvanb) [reply src/vantage6_strongaya_general/general_statistics.py:599]: Ah right, I guess there is a change in v6-tools-rdf. I didn't know the tools depended on the rdf tools, but then this makes sense

**Review by bartvanb — APPROVED** [review]
Looks good. I am not familiar enough with the code to really understand the full context but this fix looks definitely not harmful security-wise.

One small piece of advice: name branches more informatively. You may know now what `fix-12` is but it's a good habit to get into as early as possible that will help yourself or your colleagues later in life.

File: src/vantage6_strongaya_general/general_statistics.py (hunk 2 above)
Comment (JHogenboom) [reply src/vantage6_strongaya_general/general_statistics.py:599]: It is indeed coming from there :)

**Comment by JHogenboom** [comment]
Thank you @bartvanb! And more explanatory branch naming is a fair point, will do so in the future
